### ISSUE TITLE
Decorator syntax not parsing when using flow parser

### ISSUE THREAD

**Issue by Cuihongsen** [prettier/prettier#4126]
**Prettier 1.11.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEB6AVAAjgZwDYCWsAtACYE4CGARnnJgE5yVgypQRmVQDmcDmdKgA6UAgFsADhAYxMAJWasANJmCYACgFcmAYQhToCOQF9MAMwYHMAciYsYNgNyiJ02ZhwwAnnRwWrcVsAOlQiUjgAD2C-HGdRUSj3OQjzSi08OTA8Shx-AHE8CGpKPAAJZgiBKPgoUn9tPQNpKGM1UUxMSENW2AB1Ajw8AFUocQgtWAAKAEp2qE7OmAALCmCYBgIePgZFHAIALzgAUQA3Y2CwbjA4PFmXBcwTDswXgAEAEThqCagbqYAbAAGIEzF4bLY7PaHE7naZzdQvTqQKBebBwuQAXkwpAgYC04guYHs8DOximNjKABUALIAGTJsDiMwei3RFyIBBgjJgFKY+yONlUGy0cFUaTwODgLKRmAA7uEIHLguQcJJKDAwMseVM4BiZY9nlAXkw6vxZvM2Si0RwYAQbh8NZRMNiVms+DAAHIQO0Op33WVMGA6BZTWWdAA85FOXRyeU9lEJmOAXl8uGCy0q-BMAD5w69HmyI6ho3nC5gDZ0jUaQMoQBBJHboDhkKBKAwrHKNO2EC2UJRThACKRayBcjBkBKpXXqAwWABrOAwADK6rARB4yBFYpAuLAk9K05ARClsg0c544koB8lO4AVjhIgAhOdgRcrxNwOlEOA3o8TGBJC0GAACY-x3dUGFPZAQBKahfGgUdJE2fphxWZAAA4gTrZCIClPo50kGDkNwfhzlHJgAEctAIJhz0oS9ryQKcdylcQCC3BhRTrfZeDoABFLQfV-ZjDx3GAaAGUh0KQEC6w2ShBg3fRxCvGCOFaUctClKkaD7FiTBMIA)

**Input:**
```jsx
/* eslint-disable react/no-danger */
import React, { PureComponent } from 'react';
import styles from './index.less';

export default class GlobalHeader extends PureComponent {
  componentWillUnmount() {
    this.triggerResizeEvent.cancel();
  }
  
  @Debounce(600)
  triggerResizeEvent() { 
    const event = document.createEvent('HTMLEvents');
    event.initEvent('resize', true, false);
    window.dispatchEvent(event);
  }

  render() {
    const noticeData = this.getNoticeData();
    return (
      <div className={styles.header}>
      
      </div>
    );
  }
}

```
**Output:**

```
SyntaxError: Unexpected token @ (56:3)
  54 |   }
  55 | 
> 56 |   @Debounce(600)
     |   ^
  57 |   triggerResizeEvent() { // eslint-disable-line
  58 |     const event = document.createEvent('HTMLEvents');
  59 |     event.initEvent('resize', true, false);

SyntaxError: Unexpected token @ (56:3)
  54 |   }
  55 | 
> 56 |   @Debounce(600)
     |   ^
  57 |   triggerResizeEvent() { // eslint-disable-line
  58 |     const event = document.createEvent('HTMLEvents');
  59 |     event.initEvent('resize', true, false);

```

**my .prettierrc.json:**

```JSON
{
  "printWidth": 100,
  "parser": "flow",
  "singleQuote": true,
  "trailingComma": "all",
  "arrowParens": "avoid",
  "jsxBracketSameLine": true,
  "esproposal_decorators": true,
  "bracketSpacing": true
}

```

**my .eslintrc**
```JSON
{
  "parser": "babel-eslint",
  "extends": "airbnb",
  "plugins": ["prettier"],
  "env": {
    "browser": true,
    "node": true,
    "es6": true,
    "mocha": true,
    "jest": true,
    "jasmine": true
  },
  "rules": {
    "generator-star-spacing": [0],
    "consistent-return": [0],
    "react/forbid-prop-types": [0],
    "react/jsx-filename-extension": [1, { "extensions": [".js"] }],
    "global-require": [1],
    "import/prefer-default-export": [0],
    "react/jsx-no-bind": [0],
    "react/prop-types": [0],
    "react/prefer-stateless-function": [0],
    "no-else-return": [0],
    "no-restricted-syntax": [0],
    "import/no-extraneous-dependencies": [0],
    "no-use-before-define": [0],
    "jsx-a11y/no-static-element-interactions": [0],
    "jsx-a11y/no-noninteractive-element-interactions": [0],
    "jsx-a11y/click-events-have-key-events": [0],
    "jsx-a11y/anchor-is-valid": [0],
    "no-nested-ternary": [0],
    "arrow-body-style": [0],
    "import/extensions": [0],
    "no-bitwise": [0],
    "no-cond-assign": [0],
    "import/no-unresolved": [0],
    "comma-dangle": ["error", {
      "arrays": "always-multiline",
      "objects": "always-multiline",
      "imports": "always-multiline",
      "exports": "always-multiline",
      "functions": "ignore"
    }],
    "object-curly-newline": [0],
    "function-paren-newline": [0],
    "no-restricted-globals": [0],
    "require-yield": [1],
    "prettier/prettier": "error"
  },
  "parserOptions": {
    "ecmaFeatures": {
      "experimentalObjectRestSpread": true
    }
  },
  "settings": {
    "polyfills": ["fetch"]
  }
}

```

**Comment by lydell** [comment]
#2630 might be what you're looking for.

**Comment by CiGit** [comment]
The problem seems to be flow parser and decorators

**Prettier 1.11.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEACMAbAhgZx6gcQwgCMsMAJOLAEzgCdVgAdKV1VAAQBE4SIArlDBwAFADYADJICU7VDHoBLAOYqGAJTg4lALzgBRAG4IYomU1TyAvq1a2oIADQgIABxhLoOZKCz16CAB3AAV-BB8ULCMIJRpnEFwYZAAzchw4FxJ6LDAAazgYAGU3XKUoFWRFAUyQGggwVPTa8oz6GBCclQBbLCaMDJcAKxwADwAhHPzCoqxuuAAZcrh+wdcBGDcNgCZV2tL6NuQQFOIghLdlWAB1OJgAC2QADkkXS4gM65y3Y8vtBhMCXocAAjgIlMDOlgen0kGkBrUMt0lFV6DUXDoKhg4ABFAQQeB7FwwLAkW40B7IbbEnJKDDlFQAYQg3V6xyg0BWLgEGQAKqTIvCMtZrEA)
```sh
--parser flow
```

**Input:**
```jsx
 class GlobalHeader {

  @Debounce(600)
  triggerResizeEvent() { 
  }

}

```

**Output:**
```jsx
SyntaxError: Unexpected token @ (3:3)
  1 |  class GlobalHeader {
  2 | 
> 3 |   @Debounce(600)
    |   ^
  4 |   triggerResizeEvent() { 
  5 |   }
  6 | 
```

**Comment by suchipi** [comment]
As a workaround, use the babylon parser instead of the flow parser (the babylon parser supports flow types).

**Comment by scriptum** [comment]
@suchipi I'm using VS code with babylon parser (which is default), decorators cause `Unexpected token @` error (https://github.com/prettier/prettier-vscode/issues/410).

**Comment by thorn0** [comment]
fixed by #7482, not released yet
